### DISCUSSION TITLE
Got [Object object] msg when start tele bot

### DISCUSSION THREAD

**Discussion by Huybt433** [telegraf/telegraf#2014] [Q&A]
Im using nestJs, and in this function:
`@Start()
  async start(@Ctx() ctx: Context) {
await ctx.reply(
        'Welcome',
      );
}`

I got the message Welcome in my telegram but with one more line message: [Object object]. How can i config to remove [Object object]?
Thank everyone
![image](https://github.com/user-attachments/assets/6f7d1090-5394-4331-a968-dd55888f0c9b)

**Comment by suprunchuk** [comment]
Hi. Can you provide more code? For example, make a demo test project to fix the error.

**Comment by monokaijs** [comment]
Hi there, if you are facing same issue (especially with NestJs), please check any response interceptors that you are using.
I've been struggling with this for hours before knowing that NestJs Telegraf also accepts response content using return (inside service function).

Here is a sample response interceptor and how I solve the problem:
```typescript
export class ResponseInterceptor implements NestInterceptor {
  constructor(
    private readonly logger = new Logger(),
  ) {
  }

  intercept(context: ExecutionContext, next: CallHandler): Observable<unknown> {
    return next.handle().pipe(
      map((res: unknown) => this.responseHandler(res, context)),
      catchError((err: HttpException) => throwError(() => this.errorHandler(err, context))),
    );
  }

  errorHandler(exception: HttpException, context: ExecutionContext) {
    if ((context.getType() as string) === 'telegraf') {
      // handle telegraf exception
    };
  }

  async responseHandler(res: any, context: ExecutionContext) {
    if ((context.getType() as string) === 'telegraf') return res;
    // process as normal
  }
}
```

**Comment by igun997** [comment]
same with me

**Comment by gnomedevreact** [comment]
In my case the issue was caused by returning ctx.wizard.next() from a wizard step.
In nestjs-telegraf, the returned value of a handler may be treated as a reply. Since ctx.wizard.next() returns an object, it was implicitly converted to a string and sent to the chat as [object Object].

Incorrect
```
@WizardStep(0)
async step0(@Ctx() ctx: Scenes.WizardContext) {
  await ctx.reply('Hello');
  return ctx.wizard.next(); // returns an object
}
```

Correct
```
@WizardStep(0)
async step0(@Ctx() ctx: Scenes.WizardContext) {
  await ctx.reply('Hello');
  ctx.wizard.next(); // no return value
}
```
